### PR TITLE
Add request to auditTrail

### PR DESCRIPTION
**Pull request by VictorTBX**
Co-authored-by: Cassio Oliveira Silva <cassiointw1993@gmail.com>
Co-authored-by: Bruno Cavalcante Razera <brunocrazera@gmail.com>
Co-authored-by: Sander Iwase <sanderiw@gmail.com>
Co-authored-by: Matheus Vicente <matheus.vicente10@outlook.com.br>
Co-authored-by: Diogo Oliveira <diogooliveira792@gmail.com>
Co-authored-by: Julia Dibo <juliadibo@gmail.com>

### CODE REVIEW

**Comment by VictorTBX** [comment]
#66

**Comment by VictorTBX** [comment]
#56

**Comment by dalssoft** [comment]
Can we have more tests for when a request field is a entity type?

**Comment by VictorTBX** [comment]
> Can we have more tests for when a request field is a entity type?

@dalssoft do you think that one test with a the request being an entity type is enough? Do you see any other scenario. The string serialization should work for any kind of object.

**Comment by VictorTBX** [comment]
@dalssoft changes made! Could you check if are these enough and correct, please?

File: test/ifElse/ifElseStep.js [hunk 1]
```diff
@@ -188,6 +217,93 @@ describe('If Else step', () => {
...
+                                return: { Ok: '' },
+                                type: "step"
+                            },
+                            elapsedTime: uc._auditTrail.steps[0].elapsedTime,
+                        },
+                    ],
+                })
+            })
+
+            it('should audit the entity inside the entity cicularly', async () => {
```
Comment (dalssoft) [inline test/ifElse/ifElseStep.js:266]: why this test is on `ifElseStep` file? How entity type on request is related to if/else step?

File: src/usecase.js [hunk 2]
```diff
@@ -1,7 +1,10 @@
 const { step } = require('./step')
 const { Err } = require('./results')
 const { schema } = require('./schema')
+const { v4: uuidv4 } = require('uuid')
```
Comment (dalssoft) [inline src/usecase.js:4]: pls, check if you are working on the correct version. a more recent version removed uuid pkg dependency.

**Review by dalssoft — CHANGES_REQUESTED** [review]

File: test/ifElse/ifElseStep.js (hunk 1 above)
Comment (VictorTBX) [reply test/ifElse/ifElseStep.js:266]: I followed the place that was testing usecase with request, but I agree with you, these tests should be reviewed

image

File: src/usecase.js (hunk 2 above)
Comment (VictorTBX) [reply src/usecase.js]: Fixed!

File: test/ifElse/ifElseStep.js (hunk 1 above)
Comment (VictorTBX) [reply test/ifElse/ifElseStep.js]: Changes made, there was a place with a better fit for these tests afterall, thanks for the review.

File: test/usecase/simplestUsecase.js [hunk 3]
```diff
@@ -420,6 +428,83 @@ describe('A use case', () => {
             assert.ok(ret.isErr)
             assert.deepStrictEqual(ret.err, { request: [{ notDefined: true }] })
         })
+
+        describe('as an entity', () => {
```
Comment (dalssoft) [inline test/usecase/simplestUsecase.js:432]: there is already a section on this file `should audit` (line 331). all the scenarios about audit should be there.

**Review by dalssoft — CHANGES_REQUESTED** [review]

**Comment by dalssoft** [comment]
after all the required changes are submitted, pls change the doc as well: https://github.com/herbsjs/herbsjs.github.io/issues/104

we will merge both PRs together.

File: test/usecase/simplestUsecase.js (hunk 3 above)
Comment (VictorTBX) [reply test/usecase/simplestUsecase.js:432]: I think the wat I put gives us a better reading of the test context.

Look:

This way:
"A use case" + "the simplest use case with request" + **"as an entity"** + "should audit" + "tests"

The way you mentioned:
"A use case" + "the simplest use case with request" + "should audit" + "tests"

This way we create a section to encourage others to test requests as entities in the future.

What do you think?

**Comment by VictorTBX** [comment]
https://github.com/herbsjs/herbsjs.github.io/pull/108

DOC PR

**Review by dalssoft — APPROVED** [review]

**Pull request merged**